subsys/bluetooth/host/iso.c

```diff
@@ -93,7 +93,9 @@ void hci_iso(struct net_buf *buf)
 	struct bt_conn *iso;
 	uint8_t flags;
 
-	BT_DBG("buf %p", buf);
+	if (IS_ENABLED(CONFIG_BT_DEBUG_ISO_DATA)) {
+		BT_DBG("buf %p", buf);
+	}
```

Comment:
To reduce noise, conditionally define a `BT_ISO_DATA_DBG`  in https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/common/log.h
```suggestion
	BT_ISO_DATA_DBG("buf %p", buf);
```

Reply:
I'd argue that this is outside the scope of this PR. Several debug statements are already guarded by `CONFIG_BT_DEBUG_ISO_DATA`, and this PR simply fixes the remaining that wasn't added the first time. 

w.r.t your request, I think I'd need input from more, as I slightly disagree: IMO it wouldn't reduce noise, but rather readability.

Reply:
I could only find two early use of `CONFIG_BT_DEBUG_ISO_DATA` in upstream main branch. This PR adds 5 new additional conditional logging.

It is cleaner to use a macro instead of adding more lines of code for each conditional logging being done in this PR.

Reply:
Alright. A similar request has been made for https://github.com/zephyrproject-rtos/zephyr/pull/50662, so I'll go ahead and add it :)